logp/logger.go

```diff
@@ -81,62 +84,70 @@ func NewInMemory(selector string, encCfg zapcore.EncoderConfig) (*Logger, *bytes
 	return logger, &buff
 }
 
+func (l *Logger) do(f func()) {
+	if l.limiter != nil {
+		l.limiter.Do(f)
+		return
+	}
+	f()
+}
+
```

Comment:
Looks like this wrapper is messing with the depth of the stacktraces, I have to address this before merging.